### PR TITLE
Updated Release Notes.

### PR DESCRIPTION
**Pull request by apop5**
## Description
Update release notes for 202405.

- [ ] Impacts functionality?
- [ ] Impacts security?
- [ ] Breaking change?
- [ ] Includes tests?
- [x] Includes documentation?

## How This Was Tested
N/A

## Integration Instructions
N/A

### CODE REVIEW

**Comment by apop5** [comment]
@os-d @kuqin12 
Please take a look because I though there were more changes than what I captured.

File: ReadMe.rst [hunk 1]
```diff
@@ -18,44 +18,33 @@ Branch Status - release/202405
...
+
 Breaking Changes-dev
 --------------------
+- ArmPlatformPkg\Drivers\Apei\HestDxe\HestDxe.inf has been dropped.
+- ArmPlatformPkg\Drivers\HestMmErrorSources\HestErrorSourceDxe.inf has been dropped.
+- ArmPlatformPkg\Include\Protocol\HestErrorSourceInfo.h has been dropped.
+- ArmPlatformPkg\Include\Protocol\HestTable.h has been dropped.
+- ArmVirtPkg\Include\Library\FdtSerialPortAddressLib.h has been dropped.
+- ArmVirtPkg\Library\FdtSerialPortAddressLib\FdtSerialPortAddressLib.inf has been dropped.
+- ArmVirtPkg\Library\PlatformBootManagerLib\PlatformBootManagerLib.inf has been dropped.
```
Comment (kuqin12) [inline ReadMe.rst]: From my side, the extended functionality from `ArmGicLib` that used to be in 2311 is moved its own library `MsCorePkg/Library/MuArmGicExLib` in mu_plus.

**Review by kuqin12 — APPROVED** [review]

File: ReadMe.rst (hunk 1 above)
Comment (apop5) [reply ReadMe.rst]: Thanks, it should be captured now.

File: ReadMe.rst [hunk 2]
```diff
@@ -18,44 +18,35 @@ Branch Status - release/202405
...
+- ArmPlatformPkg/Include/Protocol/HestErrorSourceInfo.h has been dropped.
+- ArmPlatformPkg/Include/Protocol/HestTable.h has been dropped.
+- ArmVirtPkg/Include/Library/FdtSerialPortAddressLib.h has been dropped.
+- ArmVirtPkg/Library/FdtSerialPortAddressLib/FdtSerialPortAddressLib.inf has been dropped.
+- ArmVirtPkg/Library/PlatformBootManagerLib/PlatformBootManagerLib.inf has been dropped. 
+- ArmPkg/Drivers/ArmGic/ArmGicLib.inf has changed. Modifications for ArmGicV3SendNsG1Sgi have been moved to mu_plus MsCorePkg/Library/MuArmGicExLib.inf.
 
 Main Changes-dev
 ----------------
+- Improvements to DynamicsTablePkg.
```
Comment (makubacki) [inline ReadMe.rst]:
```suggestion
- Improvements to DynamicTablePkg.

```

**Review by makubacki — APPROVED** [review]

**Review by os-d — APPROVED** [review]

**Pull request merged**